### PR TITLE
Add support for Kaniko and JIB

### PR DESCRIPTION
**Pull request by yahavi**
- [x] All [tests](https://ci.appveyor.com/project/jfrog-ecosystem/build-info) passed. If this feature is not already covered by the tests, I added new tests.
-----

Follow up #477:
* Add support for JIB image files.
* Fix some issues discovered during development.
* Refactor Docker tests.
* Refactor DownloadResponse to return the headers.

### CODE REVIEW

File: build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/BuildDockerCreate.java [hunk 1]
```diff
@@ -78,77 +105,125 @@ public static void main(String[] ignored) {
...
+                }
+                JsonNode imageName = jsonNode.get("image");
+                if (imageName == null || imageName.isNull()) {
+                    throw new RuntimeException("Missing \"image\" in file: \"" + jibImageFiles + "\"");
+                }
+                JsonNode imageDigest = jsonNode.get("imageDigest");
+                if (imageDigest == null || imageDigest.isNull()) {
+                    throw new RuntimeException("Missing \"imageDigest\" in file: \"" + jibImageFiles + "\"");
+                }
+                imageFilesWithDigests.add(new ImageFileWithDigest(imageName.asText(), imageDigest.asText()));
```
Comment (cwholmes) [inline build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/BuildDockerCreate.java:207]: Should this be adding a image file instance for each of the tags in the file?

File: build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/BuildDockerCreate.java [hunk 2]
```diff
@@ -1,56 +1,71 @@
...
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
+import static org.jfrog.build.extractor.docker.DockerUtils.createMapper;
 import static org.jfrog.build.extractor.packageManager.PackageManagerUtils.createArtifactoryClientConfiguration;
 
 public class BuildDockerCreate extends PackageManagerExtractor {
```
Comment (eyalbe4) [inline build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/BuildDockerCreate.java:36]: BuildDockerCreate --> DockerBuildCreator

File: build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/BuildDockerCreate.java [hunk 3]
```diff
@@ -1,56 +1,71 @@
...
         this.artifactProperties = artifactProperties;
+        this.sourceRepo = sourceRepo;
+        this.imageFileType = imageFileType;
+        this.imageFile = imageFile;
+        this.logger = logger;
     }
 
     /**
-     * Allow running docker push using a new Java process.
+     * Allow running build docker create using a new Java process.
```
Comment (eyalbe4) [inline build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/BuildDockerCreate.java:68]: Allow running build docker create using a new Java process.
-->
Allow creating build-info for a published docker image in a new Java process.

File: build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/BuildDockerCreate.java [hunk 4]
```diff
@@ -60,15 +75,27 @@ public static void main(String[] ignored) {
...
-                    kanikoHandler.getImageFile(),
+                    imageFile,
+                    imageFileType,
                     ArrayListMultimap.create(clientConfiguration.publisher.getMatrixParams().asMultimap()),
                     clientConfiguration.publisher.getRepoKey(),
                     clientConfiguration.getLog());
 
-            // Exe docker push & collect build info.
+            // Exe build-docker-create & collect build info.
             dockerBuildCreate.executeAndSaveBuildInfo(clientConfiguration);
```
Comment (eyalbe4) [inline build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/BuildDockerCreate.java:99]: executeAndSaveBuildInfo --> createAndPublish

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/BuildDockerCreate.java (hunk 1 above)
Comment (yahavi) [reply build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/BuildDockerCreate.java]: Unlike Kaniko, the image file of JIB contains only one image. It is concluded from the [description](https://github.com/GoogleContainerTools/jib/blob/master/jib-maven-plugin/README.md#extended-usage) of the jib-image.json:
> The path of the image metadata json file written out during the build. Relative paths are resolved relative to the project root.

To support multiple JIB image files, we allow providing a Wildcard pattern. For example, the default pattern is: `*/target/jib-image.json`.

File: build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/BuildDockerCreate.java (hunk 4 above)
Comment (yahavi) [reply build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/BuildDockerCreate.java]: The `executeAndSaveBuildInfo` is a method of `PackageManagerExtractor`, the parent class.

**Pull request merged**